### DISCUSSION TITLE
MacOS behaviour when launching Wezterm by passing the executable pathname of a program?

### DISCUSSION THREAD

**Discussion by cormullion** [wezterm/wezterm#3787] [Q&A] [answered]
Mostly just looking for understanding here   - the answer is probably tucked away in the docs but I haven't found it.

When I launch Wezterm by passing an executable pathname, it doesn't run. (I'm double-clicking an application in the Finder which is doing `open` with an executable pathname.) An empty window opens OK with the shell prompt.

When I do this for the second (and all subsequent) times, when Wezterm is already open, the new, second (or whatever) window is created and then the program runs OK.

So, is there a configuration option that allows the second behaviour (automatically run program in new window) without having to launch Wezterm and create a window first...?

Thanks for any explanations!

**Comment by wez** [comment]
This is a rather vague description; I don't really know what you're trying to do (your higher level goal), or how you're going about it (the specific steps you're taking), so it's impossible to make any suggestions.

**Comment by cormullion** [comment]
Sorry, I'll have another go.

I have an executable file, say it's called `f`.

```plain
$ file f
f: Mach-O 64-bit executable x86_64
```

Then, in the MacOS Finder, I specify that this file should be opened in Wezterm:

<img width="404" alt="Screenshot 2023-05-27 at 17 43 57" src="https://github.com/wez/wezterm/assets/52289/0c974e0b-e662-4948-94dc-1a36976f9225">

Now I double-click this `f` file while looking at its icon in a Finder window.

*Currently* Wezterm launches (if it's not already running) but doesn't run the `f` file.
But, if Wezterm is running (there's at least one window open), so Wezterm creates a new window and runs the `f` file.
So, currently, to run this program in Wezterm, the procedure is:

1. Launch Wezterm. Check that a new window has been created.
2. Switch back to the Finder.
3. Double-click the `f` file's icon.
4. Wait for Wezterm to open a new window and run the `f` file in the new window.

I'm wondering how to configure Wezterm so that I can follow this procedure instead:

1. In the Finder.
2. Double-click the `f` file.
3. Wezterm launches, opens a new window (the first one), and runs the `f` file in this window.

Is that better?

**Reply by wez** [reply, marked as answer]
It's tricky because macOS is a bit weird.
Right now, that open-with-file stuff is gated to only occur after wezterm has launched by this code:

https://github.com/wez/wezterm/blob/13c0ba377678d1527ee129793f6e5c339c7e6e26/window/src/os/macos/app.rs#L76

because:

https://github.com/wez/wezterm/commit/5dff4b1bfbaf91143f8b9c267de013e85d321705

So a code change would be needed to resolve this.

I personally don't have the time for this, but I'm open to reviewing a PR if you wanted to take a crack at it!

**Reply by cormullion** [reply]
Ok, thanks! I wanted to know if I'd missed a preference/setting such as a "don't create window when..." kind of thing. Perhaps some Rust coding is in my future... 🤔😀
